From def485f94b7c39d018f7c5d706ca911573c066cc Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Wed, 15 Feb 2017 11:24:05 +0800
Subject: [PATCH 1516/5242] MLK-13904-4: ARM: dts: add rpmsg audio sound card
 device note

commit  869d9a041c0946d193c5a3a9adbc5e44656a4622 from
https://source.codeaurora.org/external/imx/linux-imx.git

audio will use the fourth slot in rpmsg address space.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
Acked-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7ulp-evk.dts |   18 ++++++++++++++++--
 1 file changed, 16 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/imx7ulp-evk.dts b/arch/arm/boot/dts/imx7ulp-evk.dts
index 1202c79..bd1dc02 100644
--- a/arch/arm/boot/dts/imx7ulp-evk.dts
+++ b/arch/arm/boot/dts/imx7ulp-evk.dts
@@ -166,6 +166,19 @@
 				regulator-always-on;
 		};
 	};
+
+	rpmsg_i2s: rpmsg-i2s {
+		compatible = "fsl,imx7ulp-rpmsg-i2s";
+		/* the audio device index in m4 domain */
+		fsl,audioindex = <0> ;
+		status = "okay";
+	};
+
+	sound-rpmsg {
+		compatible = "fsl,imx-audio-rpmsg";
+		model = "rpmsg-audio";
+		cpu-dai = <&rpmsg_i2s>;
+	};
 };
 
 &iomuxc1 {
@@ -421,9 +434,10 @@
 	 * --0x9FF00000~0x9FF0FFFF: pingpong
 	 * --0x9FF10000~0x9FF1FFFF: pmic
 	 * --0x9FF20000~0x9FF2FFFF: pm
+	 * --0x9FF30000~0x9FF3FFFF: audio
 	 */
-	vdev-nums = <3>;
-	reg = <0x9FF00000 0x30000>;
+	vdev-nums = <4>;
+	reg = <0x9FF00000 0x40000>;
 	status = "okay";
 };
 
-- 
1.7.9.5

